src/dynsight/_internal/vision/video_to_frame.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING
+
+import cv2
+
+if TYPE_CHECKING:
+    import pathlib
+
+    import numpy as np
+
+
+@dataclass
+class Video:
+    """Load a video file and provides utilities.
+
+    Attributes:
+        video_path:
+            File path to the video.
+    """
+
+    video_path: pathlib.Path
+    frames: list[np.ndarray] = field(default_factory=list)
+
+    def count_frame(self) -> int:
+        """Counts the total number of frames in the video.
+
+        Opens the video file at `video_path` and retrieves the frame count
+        from the video metadata.
+
+        Returns:
+            The number of frames in the video.
+
+        Raises:
+            ValueError:
+                If the video cannot be loaded.
+        """
+        capture = cv2.VideoCapture(str(self.video_path))
+        if not capture.isOpened():
+            msg = f"Impossible to load the video: {self.video_path}"
+            raise ValueError(msg)
+        count = int(capture.get(cv2.CAP_PROP_FRAME_COUNT))
+        capture.release()
+        return count
+
+    def resolution(self) -> tuple[int, int]:
+        """Retrieves the width and height of the video frames.
+
+        Opens the video file at `video_path`, reads its properties, and returns
+        the frame width and height in pixels. Raises a ValueError if the video
+        cannot be loaded.
+
+        Returns:
+            A tuple `(width, height)` representing the frame dimensions.
+
+        Raises:
+            ValueError:
+                If the video cannot be loaded.
+        """
+        capture = cv2.VideoCapture(str(self.video_path))
```

Comment:
If this is something that slows with size/resolution. Should we provide an option to open it once and parse it to each method?